databroker/broker.py

```diff
@@ -0,0 +1,448 @@
+from __future__ import print_function
+import warnings
+import six  # noqa
+from collections import deque, defaultdict
+
+import tzlocal
+import logging
+import numbers
+from .core import (Header, _inspect_descriptor,
+                   get_events as _get_events,
+                   get_table as _get_table,
+                   restream as _restream,
+                   process as _process, Images)
+
+
+# Toolz and CyToolz have identical APIs -- same test suite, docstrings.
+try:
+    from cytoolz.dicttoolz import merge
```

Comment:
Ha.

Background: I introduced this dependency fairly recently, viewing it as the first of *many* places in our databroker/datamuxer codebase that we could be using toolz/cytoolz for simplified and faster code. If we don't find more than one place to use it, sure, but I think more will come up, especially as we start dealing more often with multiple event streams.

Watching Matthew Rocklin's talk sold me on the value of the library. It implements functional patterns common in many languages -- nothing clever here -- and, like boltons, it could have been part of the standard lib.